products/appengine/terraform.yaml

```diff
@@ -70,6 +70,62 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           service_name: "ae-service"
         test_env_vars:
           org_id: :ORG_ID
+  FlexibleAppVersion: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: ["apps/{{project}}/services/{{service}}/versions/{{version_id}}"]
+    mutex: "apps/{{project}}"
+    error_retry_predicates: ["isAppEngineRetryableError"]
+    parameters:
+      service: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+        required: false
+    virtual_fields:
+      - !ruby/object:Provider::Terraform::VirtualFields
+        name: 'noop_on_destroy'
+        description: |
+          If set to `true`, the application version will not be deleted.
+      - !ruby/object:Provider::Terraform::VirtualFields
+        name: 'delete_service_on_destroy'
+        description: |
+          If set to `true`, the service will be deleted if it is the last version.    
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      custom_delete: templates/terraform/custom_delete/appversion_delete.go.erb
+      test_check_destroy: templates/terraform/custom_check_destroy/appengine.go.erb
+      encoder: templates/terraform/encoders/flex_app_version.go.erb
+    properties:
+      id: !ruby/object:Overrides::Terraform::PropertyOverride
+        name: 'version_id'
+      deployment: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      entrypoint: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      envVariables: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      threadsafe: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+      instanceClass: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true   
+      runtimeApiVersion: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      automaticScaling.maxConcurrentRequests: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      betaSettings: !ruby/object:Overrides::Terraform::PropertyOverride
```

Comment:
Are there default `betaSettings` that come from the api?

Reply:
Yes, there are, `use_deployment_manager`, `no_appserver_affinity`, and `has_docker_image` that I've seen.

Reply:
This is confusing - those values are returned in the response, but if I do an update (which is just a `POST` for a new version), it says those values (`no_appserver_affinity` and `use_deployment_manager`) aren't allowed to be set in the `flexible` environment.  Which makes me think we should `ignore_read` these, however, if there aren't conflicts with other fields, we probably want to write them to state.

Reply:
Can it be set by the user at all? If they can't be sent in a Create or Update then I would just remove the field completely. We can always add them back in as fully computed if a user needs this as an output. Growing a resource is much easier than shrinking it.